lib/polarshift/request.rb

```diff
@@ -307,6 +305,22 @@ def update_caseruns(project_id, run_id, updates)
         )
       end
 
+      def update_caseruns_smart(project_id, run_id, updates, timeout: 360)
+        res = nil
+        success = wait_for(timeout, interval: 30) {
+          res = update_caseruns(project_id, run_id, updates)
+          if res[:success]
+            return res
+          else
+            logger.info %Q{got status "#{res[:exitstatus]}" updating test run "#{run_id}" in project "#{project_id}":\n#{res[:response]}}
+            false
```

Comment:
In this case we should just leave `false` as the last statement. Return will return from the whole method. But this is Ruby madness so no worries. I fixed it. 

Thanks for implementing this feature!